core/unit_test/TestViewPadding.hpp

```diff
@@ -0,0 +1,618 @@
+auto get_subview(
+    Kokkos::View<dim_t, layout> &view,
+    [[maybe_unused]] typename std::enable_if_t<
+        std::is_same<viewDim, Kokkos::Impl::ViewDimension<ARG_5>>::value, void>
+        *ptr = nullptr) {
+  auto pair = pair_t(SUBRANGE_DIM);
+  return Kokkos::subview(view, pair, Kokkos::ALL, Kokkos::ALL, Kokkos::ALL,
+                         Kokkos::ALL);
+}
+
+template <typename dim_t, typename layout, typename viewDim>
+auto get_subview(
+    Kokkos::View<dim_t, layout> &view,
+    [[maybe_unused]] typename std::enable_if_t<
+        std::is_same<viewDim, Kokkos::Impl::ViewDimension<ARG_6>>::value, void>
+        *ptr = nullptr) {
+  auto pair = pair_t(SUBRANGE_DIM);
+  return Kokkos::subview(view, pair, Kokkos::ALL, Kokkos::ALL, Kokkos::ALL,
+                         Kokkos::ALL, Kokkos::ALL);
+}
+
+template <typename dim_t, typename layout, typename viewDim>
+auto get_subview(
+    Kokkos::View<dim_t, layout> &view,
+    [[maybe_unused]] typename std::enable_if_t<
+        std::is_same<viewDim, Kokkos::Impl::ViewDimension<ARG_7>>::value, void>
+        *ptr = nullptr) {
+  auto pair = pair_t(SUBRANGE_DIM);
+  return Kokkos::subview(view, pair, Kokkos::ALL, Kokkos::ALL, Kokkos::ALL,
+                         Kokkos::ALL, Kokkos::ALL, Kokkos::ALL);
+}
+
+template <typename dim_t, typename viewDim, typename layout,
+          typename allow_padding, typename... Vals>
+typename std::enable_if_t<
+    std::is_same<viewDim, Kokkos::Impl::ViewDimension<ARG_0>>::value ||
+        std::is_same<viewDim, Kokkos::Impl::ViewDimension<ARG_1>>::value,
+    bool>
+padding_subview_and_copy(int rank, Vals... params) {
+  bool result  = true;
+  using view_t = Kokkos::View<dim_t, layout>;
+  auto alloc_prop =
+      Kokkos::view_alloc("vDim" + std::to_string(rank), Kokkos::AllowPadding);
+  view_t v(alloc_prop, params...);
+  auto v_sub        = get_subview<dim_t, layout, viewDim>(v);
+  auto v_sub_mirror = Kokkos::create_mirror_view(v_sub);
+  result &= ASSERT_VIEW_STRIDES(v, v_sub, rank);
+  /* We never pad for rank 0 and rank 1 and for LR && dyn_rank > 1 */
+  result &= ASSERT_VIEW_STRIDES(v_sub, v_sub_mirror, rank);
+  return result;
+}
+
+template <typename dim_t, typename viewDim, typename layout,
+          typename allow_padding, typename... Vals>
+typename std::enable_if_t<
+    !std::is_same<viewDim, Kokkos::Impl::ViewDimension<ARG_0>>::value &&
+        !std::is_same<viewDim, Kokkos::Impl::ViewDimension<ARG_1>>::value,
+    bool>
+padding_subview_and_copy(int rank, Vals... params) {
+  bool result  = true;
+  using view_t = Kokkos::View<dim_t, layout>;
+  auto alloc_prop =
+      Kokkos::view_alloc("vDim" + std::to_string(rank), Kokkos::AllowPadding);
+  view_t v(alloc_prop, params...);
+  auto v_sub        = get_subview<dim_t, layout, viewDim>(v);
+  auto v_sub_mirror = Kokkos::create_mirror_view(v_sub);
+  result &= ASSERT_VIEW_STRIDES(v, v_sub, rank);
+  if (v_sub_mirror.span_is_contiguous())
```

Comment:
Do we need to test that `v_sub.span_is_contiguous() == v_sub_mirror.span_is_contiguous()`? It seems like relying on the mirror having that correct is an unjustified assumption about the code exercised by the test.

Reply:
We are checking here that a mirror of a subview that is contiguous in memory has different strides as a result of a different size. This must happen if the subview is a slice among the first significant dimension. We do this to guarantee that the addition of padding still allows contiguous subwies.